src/apiHelper.ts

```diff
@@ -14,37 +18,51 @@ export async function getWeatherData(city : CityInterface): Promise<[CurrentWeat
 
 	const url = `${baseUrl}?${params.toString()}`;
 
-	try {
-		const response = await fetch(url);
+	const timeInterval$ = interval(600000); 
 
-		if (!response.ok) {
-			throw new Error('Unable to get response');
-		}
-		const data : WeatherDataResponseInterface = await response.json();
-		if (data?.current?.temperature_2m && data?.current?.temperature_2m && data?.daily?.temperature_2m_max.length>0 && data?.daily?.temperature_2m_min.length>0) {
-			const currentWeatherData: CurrentWeatherDataInterface = {
-				temperature : data.current.temperature_2m,
-				temperatureMax: data.daily.temperature_2m_max[0],
-				temperatureMin: data.daily.temperature_2m_min[0],
-				weatherCode: data.current.weather_code,
-			};
+	return timeInterval$.pipe(
+		startWith(0),
+		switchMap(() => 
+			fromFetch(url).pipe(
+			switchMap(response => fromPromise<WeatherDataResponseInterface>(response.json())),
+			switchMap(data => {
+				if (
+					!data?.current?.temperature_2m &&
+					!data?.current?.weather_code &&
+					!(data?.daily?.temperature_2m_max?.length > 0) &&
+					!(data?.daily?.temperature_2m_min?.length > 0)
+				) {
+					throw new Error('Data is not complete or valid');
```

Comment:
You can potentially `console.error` here and then `return EMPTY;` if you like. Otherwise, you have to have some kind of catch on your observable pipeline

Reply:
Nevermind you do, it was just way down